File: src/main/java/com/github/michael72/pumlsrv/Style.java
package com.github.michael72.pumlsrv;

public class Style {
    private static final String header = "skinparam {\n" +
    "}\n" ; 
    private static final String light = 
        "skinparam {\n" + 
        "Shadowing false\n" +
        "}\n" + 
        "!define STYLE_ACCENT #81D4FA\n" + 
        "!define STYLE_BGC #FFF\n" + 
        "!define STYLE_BGC2 #EEE\n" + 
        "!define STYLE_FGC #000\n";
    
    private static final String dark = 
        "!define STYLE_ACCENT #81D4FA\n" +
        "!define STYLE_BGC #203562\n" +
        "!define STYLE_BGC2 #1C2A38\n" + 
        "!define STYLE_FGC #FFF\n" + 
        "hide circle\n";
    
    private static final String modern = 
        "skinparam {\n" +
        "defaultFontColor STYLE_FGC\n" +
        "ArrowColor STYLE_ACCENT\n" +
        "ActivityBorderColor STYLE_ACCENT\n" +
        "ActivityDiamondBorderColor STYLE_ACCENT\n" +
        "ActorBorderColor STYLE_ACCENT\n" +
        "AgentBorderColor STYLE_ACCENT\n" +
        "ArtifactBorderColor STYLE_ACCENT\n" +
        "BoundaryBorderColor STYLE_ACCENT\n" +
        "ClassBorderColor STYLE_ACCENT\n" +
        "CloudBorderColor STYLE_ACCENT\n" +
        "CollectionsBorderColor STYLE_ACCENT\n" +
        "ComponentBorderColor STYLE_ACCENT\n" +
        "ControlBorderColor STYLE_ACCENT\n" +
        "DatabaseBorderColor STYLE_ACCENT\n" +
        "EntityBorderColor STYLE_ACCENT\n" +
        "FileBorderColor STYLE_ACCENT\n" +
        "FolderBorderColor STYLE_ACCENT\n" +
        "FrameBorderColor STYLE_ACCENT\n" +
        "InterfaceBorderColor STYLE_ACCENT\n" +
        "LegendBorderColor STYLE_ACCENT\n" +
        "NodeBorderColor STYLE_ACCENT\n" +
        "NoteBorderColor STYLE_ACCENT\n" +
        "ObjectBorderColor STYLE_ACCENT\n" +
        "PackageBorderColor STYLE_ACCENT\n" +
        "ParticipantBorderColor STYLE_ACCENT\n" +
        "PartitionBorderColor STYLE_ACCENT\n" +
        "QueueBorderColor STYLE_ACCENT\n" +
        "RectangleBorderColor STYLE_ACCENT\n" +
        "SequenceBoxBorderColor STYLE_ACCENT\n" +
        "SequenceDividerBorderColor STYLE_ACCENT\n" +
        "SequenceGroupBorderColor STYLE_ACCENT\n" +
        "SequenceLifeLineBorderColor STYLE_ACCENT\n" +
        "SequenceReferenceBorderColor STYLE_ACCENT\n" +
        "StackBorderColor STYLE_ACCENT\n" +
        "StateBorderColor STYLE_ACCENT\n" +
        "StorageBorderColor STYLE_ACCENT\n" +
        "SwimlaneBorderColor STYLE_ACCENT\n" +
        "UsecaseBorderColor STYLE_ACCENT\n" +
        "BackgroundColor STYLE_BGC\n" +
        "ActivityBackgroundColor STYLE_BGC\n" +
        "ActivityDiamondBackgroundColor STYLE_BGC\n" +
        "ActorBackgroundColor STYLE_BGC\n" +
        "AgentBackgroundColor STYLE_BGC\n" +
        "ArtifactBackgroundColor STYLE_BGC\n" +
        "BoundaryBackgroundColor STYLE_BGC\n" +
        "ClassBackgroundColor STYLE_BGC\n" +
        "ClassHeaderBackgroundColor STYLE_BGC\n" +
        "CloudBackgroundColor STYLE_BGC\n" +
        "CollectionsBackgroundColor STYLE_BGC\n" +
        "ComponentBackgroundColor STYLE_BGC\n" +
        "ControlBackgroundColor STYLE_BGC\n" +
        "DatabaseBackgroundColor STYLE_BGC\n" +
        "EntityBackgroundColor STYLE_BGC\n" +
        "FileBackgroundColor STYLE_BGC\n" +
        "FolderBackgroundColor STYLE_BGC\n" +
        "FrameBackgroundColor STYLE_BGC\n" +
        "IconPackageBackgroundColor STYLE_BGC\n" +
        "IconPrivateBackgroundColor STYLE_BGC\n" +
        "IconProtectedBackgroundColor STYLE_BGC\n" +
        "IconPublicBackgroundColor STYLE_BGC\n" +
        "InterfaceBackgroundColor STYLE_BGC\n" +
        "LegendBackgroundColor STYLE_BGC\n" +
        "NoteBackgroundColor STYLE_BGC2\n" +
        "ObjectBackgroundColor STYLE_BGC\n" +
        "PackageBackgroundColor STYLE_BGC\n" +
        "ParticipantBackgroundColor STYLE_BGC\n" +
        "PartitionBackgroundColor STYLE_BGC2\n" +
        "QueueBackgroundColor STYLE_BGC\n" +
        "RectangleBackgroundColor STYLE_BGC\n" +
        "SequenceBoxBackgroundColor STYLE_BGC2\n" +
        "SequenceDividerBackgroundColor STYLE_BGC\n" +
        "SequenceGroupBackgroundColor STYLE_BGC\n" +
        "SequenceGroupBodyBackgroundColor STYLE_BGC\n" +
        "SequenceLifeLineBackgroundColor STYLE_BGC\n" +
        "SequenceReferenceBackgroundColor STYLE_BGC\n" +
        "SequenceReferenceHeaderBackgroundColor STYLE_BGC\n" +
        "StackBackgroundColor STYLE_BGC\n" +
        "StateBackgroundColor STYLE_BGC\n" +
        "StereotypeABackgroundColor STYLE_BGC\n" +
        "StereotypeCBackgroundColor STYLE_BGC\n" +
        "StereotypeEBackgroundColor STYLE_BGC\n" +
        "StereotypeIBackgroundColor STYLE_BGC\n" +
        "StereotypeNBackgroundColor STYLE_BGC\n" +
        "StorageBackgroundColor STYLE_BGC\n" +
        "TitleBackgroundColor STYLE_BGC\n" +
        "UsecaseBackgroundColor STYLE_BGC\n" +

        "roundcorner 6\n" +
        "NoteFontName  Gill Sans Ultra Bold Condensed\n" +
        "defaultFontName Malgun Gothic\n" + 
        "defaultFontSize 16\n" +
        "}\n";

    public static String darkTheme() {
        return header + dark + modern +         
            "skinparam BackgroundColor 202020\n" +
            "skinparam defaultFontColor e0f0ff\n";       
    }
    public static String lightTheme() {
        return header + light + modern;       
    }

}

File: src/main/kotlin/com/github/michael72/pumlsrv/Style.kt
package com.github.michael72.pumlsrv

object Style {
    private const val HEADER = """
        skinparam {
        }
        """

    private const val LIGHT = """
        skinparam {
        Shadowing false
        }
        !define STYLE_ACCENT #81D4FA
        !define STYLE_BGC #FFF
        !define STYLE_BGC2 #EEE
        !define STYLE_FGC #000
        """

    private const val DARK = """
        !define STYLE_ACCENT #81D4FA
        !define STYLE_BGC #203562
        !define STYLE_BGC2 #1C2A38
        !define STYLE_FGC #FFF
        hide circle
        """

    private const val MODERN = """
        skinparam {
        defaultFontColor STYLE_FGC
        ArrowColor STYLE_ACCENT
        ActivityBorderColor STYLE_ACCENT
        ActivityDiamondBorderColor STYLE_ACCENT
        ActorBorderColor STYLE_ACCENT
        AgentBorderColor STYLE_ACCENT
        ArtifactBorderColor STYLE_ACCENT
        BoundaryBorderColor STYLE_ACCENT
        ClassBorderColor STYLE_ACCENT
        CloudBorderColor STYLE_ACCENT
        CollectionsBorderColor STYLE_ACCENT
        ComponentBorderColor STYLE_ACCENT
        ControlBorderColor STYLE_ACCENT
        DatabaseBorderColor STYLE_ACCENT
        EntityBorderColor STYLE_ACCENT
        FileBorderColor STYLE_ACCENT
        FolderBorderColor STYLE_ACCENT
        FrameBorderColor STYLE_ACCENT
        InterfaceBorderColor STYLE_ACCENT
        LegendBorderColor STYLE_ACCENT
        NodeBorderColor STYLE_ACCENT
        NoteBorderColor STYLE_ACCENT
        ObjectBorderColor STYLE_ACCENT
        PackageBorderColor STYLE_ACCENT
        ParticipantBorderColor STYLE_ACCENT
        PartitionBorderColor STYLE_ACCENT
        QueueBorderColor STYLE_ACCENT
        RectangleBorderColor STYLE_ACCENT
        SequenceBoxBorderColor STYLE_ACCENT
        SequenceDividerBorderColor STYLE_ACCENT
        SequenceGroupBorderColor STYLE_ACCENT
        SequenceLifeLineBorderColor STYLE_ACCENT
        SequenceReferenceBorderColor STYLE_ACCENT
        StackBorderColor STYLE_ACCENT
        StateBorderColor STYLE_ACCENT
        StorageBorderColor STYLE_ACCENT
        SwimlaneBorderColor STYLE_ACCENT
        UsecaseBorderColor STYLE_ACCENT
        BackgroundColor STYLE_BGC
        ActivityBackgroundColor STYLE_BGC
        ActivityDiamondBackgroundColor STYLE_BGC
        ActorBackgroundColor STYLE_BGC
        AgentBackgroundColor STYLE_BGC
        ArtifactBackgroundColor STYLE_BGC
        BoundaryBackgroundColor STYLE_BGC
        ClassBackgroundColor STYLE_BGC
        ClassHeaderBackgroundColor STYLE_BGC
        CloudBackgroundColor STYLE_BGC
        CollectionsBackgroundColor STYLE_BGC
        ComponentBackgroundColor STYLE_BGC
        ControlBackgroundColor STYLE_BGC
        DatabaseBackgroundColor STYLE_BGC
        EntityBackgroundColor STYLE_BGC
        FileBackgroundColor STYLE_BGC
        FolderBackgroundColor STYLE_BGC
        FrameBackgroundColor STYLE_BGC
        IconPackageBackgroundColor STYLE_BGC
        IconPrivateBackgroundColor STYLE_BGC
        IconProtectedBackgroundColor STYLE_BGC
        IconPublicBackgroundColor STYLE_BGC
        InterfaceBackgroundColor STYLE_BGC
        LegendBackgroundColor STYLE_BGC
        NoteBackgroundColor STYLE_BGC2
        ObjectBackgroundColor STYLE_BGC
        PackageBackgroundColor STYLE_BGC
        ParticipantBackgroundColor STYLE_BGC
        PartitionBackgroundColor STYLE_BGC2
        QueueBackgroundColor STYLE_BGC
        RectangleBackgroundColor STYLE_BGC
        SequenceBoxBackgroundColor STYLE_BGC2
        SequenceDividerBackgroundColor STYLE_BGC
        SequenceGroupBackgroundColor STYLE_BGC
        SequenceGroupBodyBackgroundColor STYLE_BGC
        SequenceLifeLineBackgroundColor STYLE_BGC
        SequenceReferenceBackgroundColor STYLE_BGC
        SequenceReferenceHeaderBackgroundColor STYLE_BGC
        StackBackgroundColor STYLE_BGC
        StateBackgroundColor STYLE_BGC
        StereotypeABackgroundColor STYLE_BGC
        StereotypeCBackgroundColor STYLE_BGC
        StereotypeEBackgroundColor STYLE_BGC
        StereotypeIBackgroundColor STYLE_BGC
        StereotypeNBackgroundColor STYLE_BGC
        StorageBackgroundColor STYLE_BGC
        TitleBackgroundColor STYLE_BGC
        UsecaseBackgroundColor STYLE_BGC

        roundcorner 6
        NoteFontName  Gill Sans Ultra Bold Condensed
        defaultFontName Malgun Gothic
        defaultFontSize 16
        }
        """

    fun darkTheme(): String {
        return HEADER + DARK + MODERN +
                "skinparam BackgroundColor 202020\n" +
                "skinparam defaultFontColor e0f0ff\n"
    }

    fun lightTheme(): String {
        return HEADER + LIGHT + MODERN
    }
}